src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/mapper/ActionPlanDomainMapper.kt

```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.mapper
+
+import org.mapstruct.Mapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.ActionPlan
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ActionPlanResponse
+
+@Mapper(
+  uses = [
+    GoalDomainMapper::class,
+  ],
+)
+interface ActionPlanDomainMapper {
```

Comment:
I'm not sure about the use of / focus of the word "Domain" in the class name in these mappers .... 🤔 

For the ones I introduced in the database layer, I used the term "Entity" in the class name, because they are mapping an entity to and from the domain - eg: `GoalEntityMapper`

How about `GoalResourceMapper` or similar?

I guess the problem is that if we both used "Domain" in our mapper class names we'd have 2 `GoalDomainMapper` classes - one here in the `resource` package that is about mapping resource instances to domain instances; and one in the `entity` package that is about mapping entities to domain instances.

We don't want duplicate class names I don't think?

Reply:
Or `GoalModelMapper` given that your methods are named `fromModelToDomain` etc ?
Just a thought .... 🤔

Reply:
Good point. `GoalModelMapper` is ok, but I find `Model` a bit vague (I slightly prefer `Api`, but you could argue that's vague too). I think `GoalResourceMapper` works, since `Resource` seems to imply REST here.